contracts/instance/InstanceAccessManager.sol

```diff
@@ -1,292 +1,526 @@
+        returns (bool) 
+    {
+        return _revokeRole(roleId, member);
     }
 
-    function revokeRole(RoleId roleId, address member) external restricted() returns (bool revoked) {
+    // INSTANCE_OWNER_ROLE
+    // IMPORTANT: unbounded function, revoke all or revert
+    // Instance owner role decides what to do in case of custom role admin bening revoked:
+    // 1) revoke custom role from ALL members
+    // 2) revoke custom role admin from ALL members
+    // 3) 1) + 2)
+    // 4) revoke only 1 member of custom role admin
+    function revokeRoleAllMembers(RoleId roleId) 
+        external
+        restrictedToRoleAdmin(roleId) 
+        returns (bool revoked)
+    {
         if (!roleExists(roleId)) {
-            revert IAccess.ErrorIAccessRevokeNonexstentRole(roleId);
+            revert IAccess.ErrorIAccessRoleIdInvalid(roleId);
         }
 
-        if (EnumerableSet.contains(_roleMembers[roleId], member)) {
-            _accessManager.revokeRole(roleId.toInt(), member);
+        uint memberCount = EnumerableSet.length(_roleMembers[roleId]);
+        for(uint memberIdx = 0; memberIdx < memberCount; memberIdx++)
+        {
+            address member = EnumerableSet.at(_roleMembers[roleId], memberIdx);
             EnumerableSet.remove(_roleMembers[roleId], member);
-            return true;
-        }
-
-        return false;
+            // TODO no return value, can fail silentlly
+            _accessManager.revokeRole(roleId.toInt(), member);
+        }  
     }
 
     /// @dev not restricted function by intention
-    /// the restriction to role members is already enforced by the call to the access manger
-    function renounceRole(RoleId roleId) external returns (bool revoked) {
-        address member = msg.sender;
-
-        if (!roleExists(roleId)) {
-            revert IAccess.ErrorIAccessRenounceNonexstentRole(roleId);
+    /// the restriction to role members is already enforced by the call to the access manager
+    function renounceRole(RoleId roleId) 
+        external 
+        returns (bool) 
+    {
+        IAccess.Type rtype = _roleInfo[roleId].rtype;
+        if(rtype == IAccess.Type.Core || rtype == IAccess.Type.Gif) {
+            revert IAccess.ErrorIAccessRoleTypeInvalid(roleId, rtype);
         }
 
-        if (EnumerableSet.contains(_roleMembers[roleId], member)) {
-            // cannot use accessManger.renounce as it directly checks against msg.sender
-            _accessManager.revokeRole(roleId.toInt(), member);
-            EnumerableSet.remove(_roleMembers[roleId], member);
-            return true;
-        }
+        address member = msg.sender;
+        // cannot use accessManger.renounce as it directly checks against msg.sender
+        return _revokeRole(roleId, member);
+    }
 
-        return false;
+    function roleExists(RoleId roleId) public view returns (bool exists) {
+        return _roleInfo[roleId].createdAt.gtz();
+    }
+    // TODO returns ADMIN_ROLE id for non existent roleId
```

Comment:
why not just revert if the role doesn't exist?

Reply:
Because this is getter function, its user may not want revert